File: SDS.Clould/SDS.Clould.WebAPI/DataModels/NewTransportDataModel.cs
﻿namespace SDS.Clould.WebAPI.DataModels
{
    using System;
    using System.Collections.Generic;
    using System.ComponentModel.DataAnnotations;
    using System.Linq;

    using SDS.Clould.Models;

    public class NewTransportDataModel
    {
        public static Transport FromModel(NewTransportDataModel model, string driverId)
        {
            return new Transport()
            {
                 Arrival = model.Arrival,
                 AvailableKilograms = model.AvailableKilograms,
                 AvailableSpace = model.AvailableSpace,
                 Departure = model.Departure,
                 FromTown = model.FromTown,
                 ToTown = model.ToTown,
                 UserId = driverId,
                 AdditionalContact = model.AdditionalContact
            };
        }

        [Required]
        public string FromTown { get; set; }

        [Required]
        public string ToTown { get; set; }

        public int AvailableSpace { get; set; }

        public int AvailableKilograms { get; set; }

        public DateTime Arrival { get; set; }

        public DateTime Departure { get; set; }

        public string DriverName { get; set; }

        public string DriverPhone { get; set; }

        [Required]
        public string AdditionalContact { get; set; }
    }
}

File: SDS.Mobile/app/src/main/java/com/example/spawneddeliveryservice/models/NewTransportDataModel.java
package com.example.spawneddeliveryservice.models;

public class NewTransportDataModel {
    public String fromTown;
    public String toTown;
    public int availableSpace;
    public int availableKilograms;
    public String arrival;//2015-03-17 format;
    public String departure;//2015-03-17 format
    public String driverName;
    public String driverPhone;
    public String additionalContact;

    public NewTransportDataModel(String fromTown, String toTown, int availableSpace, int availableKilograms,
                                 String arrival, String departure, String driverName, String driverPhone, String additionalContact) {
        this.fromTown = fromTown;
        this.toTown = toTown;
        this.availableSpace = availableSpace;
        this.availableKilograms = availableKilograms;
        this.arrival = arrival;
        this.departure = departure;
        this.driverName = driverName;
        this.driverPhone = driverPhone;
        this.additionalContact = additionalContact;
    }
}
